wallet/walletrpc.c

```diff
@@ -390,23 +391,47 @@ static const struct json_command newaddr_command = {
 };
 AUTODATA(json_command, &newaddr_command);
 
-static void json_listfunds(struct command *cmd, const char *buffer,
-			   const jsmntok_t *params)
+static void json_listfunds(struct command *cmd, const char *buffer UNUSED,
+			   const jsmntok_t *params UNUSED)
 {
 	struct json_result *response = new_json_result(cmd);
+	struct peer *p;
 	struct utxo **utxos =
 	    wallet_get_utxos(cmd, cmd->ld->wallet, output_state_available);
 	json_object_start(response, NULL);
 	json_array_start(response, "outputs");
-	for (int i = 0; i < tal_count(utxos); i++) {
+	for (size_t i = 0; i < tal_count(utxos); i++) {
 		json_object_start(response, NULL);
 		json_add_txid(response, "txid", &utxos[i]->txid);
 		json_add_num(response, "output", utxos[i]->outnum);
 		json_add_u64(response, "value", utxos[i]->amount);
 		json_object_end(response);
 	}
 	json_array_end(response);
+
+	/* Add funds that are allocated to channels */
+	json_array_start(response, "channels");
+	list_for_each(&cmd->ld->peers, p, list) {
+		if (!p->our_msatoshi || !p->funding_txid)
+			continue;
+
+		json_object_start(response, NULL);
+		json_add_pubkey(response, "peer_id", &p->id);
+		if (p->scid)
+			json_add_short_channel_id(response, "short_channel_id",
+						  p->scid);
+		else
+			json_add_null(response, "short_channel_id");
+
+		json_add_u64(response, "value_sat", *p->our_msatoshi/1000);
+		json_add_u64(response, "value_total_sat",
+			     p->funding_satoshi);
```

Comment:
s/value/channel/ here?

Reply:
Mainly used `value` here since we already have it in the other array, can replace of course